hazelcast/src/main/java/com/hazelcast/sql/HazelcastSqlException.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql;
+
+import com.hazelcast.core.HazelcastException;
+
+/**
+ * Exception occurred during SQL query execution.
+ */
+public final class HazelcastSqlException extends HazelcastException {
+    /** Code of the exception. */
+    private int code;
```

Comment:
It is advised by the `checkstyle` since there are no public constructors. I do not see any problems with it for now, since we do not have any children classes. We may remove it in the future if needed.